internal/model/rwfolder.go

```diff
@@ -575,7 +575,11 @@ func (p *rwFolder) deleteFile(file protocol.FileInfo) {
 
 	realName := filepath.Join(p.dir, file.Name)
 
-	if p.versioner != nil {
+	cur, ok := p.model.CurrentFolderFile(p.folder, file.Name)
+	if ok && cur.Version.Concurrent(file.Version) {
+		// There is a conflict here. Move the file to a conflict copy instead of deleting.
+		err = osutil.InWritableDir(moveForConflict, realName)
```

Comment:
Shouldn't we try to archive if the conflicted version failed? For example, if the filename becomes to long, it would be nice to still have the old version.